### PR TITLE
Merge/foundation release/1.13.5

### PR DESCRIPTION
**Pull request by meowsbits**
Rel #575, #563. 

Merges https://github.com/ethereum/go-ethereum/releases/tag/v1.13.5.

### CODE REVIEW

**Comment by meowsbits** [comment]
Not sure why lint and tests are failing, they pass for me locally.

File: eth/ethconfig/config.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ var Defaults = Config{
 		DatasetsOnDisk:   2,
 		DatasetsLockMmap: false,
 	},
-	NetworkId:          vars.DefaultNetworkID,
+	NetworkId:          0, // enable auto configuration of networkID == chainID
```
Comment (ziogaschr) [inline eth/ethconfig/config.go:78]: Just a note to be extra cautious in this change.

Classic config has this hardcoded to 1, which is correct. :+1:

File: eth/handler.go [hunk 2]
```diff
@@ -284,7 +284,12 @@ func newHandler(config *handlerConfig) (*handler, error) {
 			}
 			return 0, nil
 		}
-		return h.chain.InsertChain(blocks)
+		n, err := h.chain.InsertChain(blocks)
+		// PTAL(meowsbits) This chunk was removed upstream. (No acceptTxs.Store...)
```
Comment (ziogaschr) [inline eth/handler.go:288]: Nice that you've added this comment here so as we don't miss it on subsequent merges.

**Review by ziogaschr — APPROVED** [review]

**Comment by meowsbits** [comment]
This branch is a superset of the 13.x branches:PRs below, with these stacked PRs originally intended only to piece-out the diffs for review's sake.
- `merge/foundation-release/1.13.4`: https://github.com/etclabscore/core-geth/pull/575
- `merge/foundation-release/1.13.1`: https://github.com/etclabscore/core-geth/pull/563

Those PRs have been reviewed and :heavy_check_mark:'d.

This branch has been merged with the latest versions of both those branches (as noops, but just to be safe), so I'm going to close those PRs in favor of this one. Then I'll change the base (/target) branch of this PR to `master`.

**Comment by meowsbits** [comment]
Don't know why [the "go generate check" is failing](https://github.com/etclabscore/core-geth/actions/runs/7978914253/job/21785080899?pr=593)... can't reproduce locally.

**Comment by meowsbits** [comment]
Gosh I just love green lights. Thanks @ziogaschr :green_heart:

**Comment by ziogaschr** [comment]
Syncing finished fine on all nodes @meowsbits. Let's merge it

**Pull request merged**